lib/Onyx.js

```diff
@@ -428,17 +460,15 @@ function set(key, val) {
  * @returns {Promise}
  */
 function multiSet(data) {
-    // AsyncStorage expenses the data in an array like:
-    // [["@MyApp_user", "value_1"], ["@MyApp_key", "value_2"]]
-    // This method will transform the params from a better JSON format like:
-    // {'@MyApp_user': 'myUserValue', '@MyApp_key': 'myKeyValue'}
-    const keyValuePairs = _.reduce(data, (finalArray, val, key) => ([
-        ...finalArray,
-        [key, JSON.stringify(val)],
-    ]), []);
+    const keyValuePairs = prepareKeyValuePairsForStorage(data);
+
+    // Capture (non-stringified) changes to cache
+    cache.merge(_.pairs(data));
+
+    // Optimistically inform subscribers
+    _.each(data, (val, key) => keyChanged(key, val));
```

Comment:
This comment might change based on the resolution to the comment above... but I think in this case since we've used a `merge()` that we are in some cases sending one copy of the `val` (object or array) to `keyChanged()` and another to the cache.

Wouldn't it be more correct to update these subscribers with the value _from the cache_ so future subscribers are referencing that same value instead of some referencing the value passed to `multiSet()` and others the cached value (retrieved via `get()`)?

Reply:
Updated to 

```js
    _.each(data, (val, key) => {
        // Update cache and optimistically inform subscribers
        cache.set(key, val);
        keyChanged(key, val);
    });
```

So there's no need for a test that reference matches in this particular case